spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumerTests.java

```diff
@@ -240,11 +243,14 @@ private void testRequeueOrNotGuts(Exception ex, boolean expectedRequeue,
 	public void testAlwaysCancelAutoRecoverConsumer() throws IOException {
 		ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
 		Connection connection = mock(Connection.class);
-		Channel channel = mock(AutorecoveringChannel.class);
+		ChannelProxy channel = mock(ChannelProxy.class);
+		Channel rabbitChannel = mock(AutorecoveringChannel.class);
+		when(channel.getTargetChannel()).thenReturn(rabbitChannel);
 
 		when(connectionFactory.createConnection()).thenReturn(connection);
 		when(connection.createChannel(anyBoolean())).thenReturn(channel);
-		when(channel.isOpen()).thenReturn(true);
+		final AtomicBoolean isOpen = new AtomicBoolean(true);
```

Comment:
I understand, but `isOpen()` is called quite often; I specifically coded it this way because we want it to always return `true` until we want it to return `false`. Your solution is brittle - perhaps it works now, but it will break if another `isOpen()` is added later.